go/internal/initutil/manager.go

```diff
@@ -267,3 +279,31 @@ func (m *Manager) AdvancedHelp() string {
 	tw.Flush()
 	return strings.TrimSpace(b.String())
 }
+
+// prepareForGetter prepare locks and ctx timeouts for an external getter.
+// it returns a cleanup and should be defered.
+func (m *Manager) prepareForGetter() func() {
+	m.mutex.Lock()
+	if m.InitialisaitonTimeout >= 0 {
+		finishChan := make(chan struct{})
+		trueCtx := m.getContext()
+		var cancel func()
+		m.ctx, cancel = context.WithCancel(trueCtx)
```

Comment:
can you check if we are using m.ctx as initializer of some parts or just for init cancellation? 

I think that you may cancel things "after" the init, even if you restore the `trueCtx`

Reply:
> I think that you may cancel things "after" the init, even if you restore the trueCtx

Who is you ?
- You is the goroutine lambda in `prepareForGetter`, it shouldn't because the `finishChan` is closed in the deffered returned lambda.
- You is the user who cancel his parrent context, it should because the ctx is pulled from the user with a `cancel`.

The only "dirty" thing is that the deffered lambda doesn't wait for the ticker watcher goroutine to be freed, I guess I can add a waitgroup but I don't think that important (at worst nothing happend and the goroutine spend some time doing nothing).

Reply:
> can you check if we are using m.ctx as initializer of some parts or just for init cancellation?

Rereading makes me understand the question, it's used as ctx evrywhere as context but it shouldn't be canceled if the deffered lambda is executed.

Reply:
👍